### ISSUE TITLE
New deployment sometimes fails with "the object has been modified" error

### ISSUE THREAD

**Issue by liggitt** [openshift/origin#5054]
```
$ oc deploy client-cert --latest
... waited a while, watched oc status, saw deployment had failed ...
$ oc logs pod/client-cert-3-deploy 
I1010 03:55:28.233063       1 deployer.go:197] Deploying from default/client-cert-2 to default/client-cert-3 (replicas: 1)
F1010 03:55:28.261981       1 deployer.go:64] couldn't assign source annotation to deployment default/client-cert-3: replicationControllers "client-cert-3" cannot be updated: the object has been modified; please apply your changes to the latest version and try again
```

unfortunately, I reran the deploy command and got a successful deployment before thinking to capture the output: https://gist.github.com/liggitt/65bd2d601ce0e3f78ca9

**Comment by ironcladlou** [comment]
Duplicate of #4678 which was already resolved. Are you sure your deployer image contains the fix from #4679?

**Comment by liggitt** [comment]
it did not. closing
